File: DataStructures/LinkedList/IterativePosition.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def push(self,new_data):
        new_node = Node(new_data)
        new_node.next = self.head
        self.head = new_node

    def GetNth(self, position):
        current = self.head
        count = 0
        while(current):
            if(count == position):
                return current.data
            count+=1
            current = current.next

        assert(false)
        return 0

if __name__ == '__main__':
    llist = LinkedList()

    llist.push(11)
    llist.push(15)
    llist.push(34)
    llist.push(23)
    llist.push(21)

    n = 3
    print("Element at index 3 is:", llist.GetNth(n))

File: DataStructures/LinkedList/IterativePosition.java
class Node 
{
    int data;
    Node next;
    Node(int d)
    {
        data = d;
    }
}

class LinkedList 
{
    Node head = null;

    public void push(int new_data)
    {
        Node new_node = new Node(new_data);
        new_node.next = head;
        head = new_node;
    }

    public int GetNth(Node head, int index)
    {
        Node current = head;
        int count = 0;
        while(current != null)
        {
            if(count == index)
            {
                return current.data;
            }
            count++;
            current=current.next;
        }
    }

    public static void main(String[] args)
    {
        LinkedList llist = new LinkedList();
        llist.push(27);
        llist.push(26);
        llist.push(25);
        llist.push(24);
        llist.push(23);
        int index = 3;

        System.out.print("The element at index " + index "is " GetNth(llist.head, index)));
        // cn also write 
        System.out.fomrat("The element at %d is %d", index, GetNth(llist.head, index));
    }
}